### PR TITLE
Add wear_keep to datalayer-helper-guide and sample

### LINKED ISSUE #2281: android_wear_capabilities and resource shrinker
See: https://issuetracker.google.com/issues/348688201

But to resume `tools:keep="@array/android_wear_capabilities" ` is not a valid way to ensure that the resource is kept when present inside the xml itself.
Docs should be amended with for example the solution from https://issuetracker.google.com/issues/348688201#comment4

### PR DESCRIPTION
**Pull request by oas004**
#### WHAT
Add wear_keep.xml to data layer helper guide and sample

From what I understand from https://issuetracker.google.com/issues/348688201#comment4 that is what we needed for resource shrinker to work. I'm not 100% sure so all feedback is more than welcome on that.

#### WHY
resolves #2281 

#### HOW
Add wear_keep.xml to data layer helper guide and sample

#### Checklist :clipboard:
- [ ] Add explicit visibility modifier and explicit return types for public declarations
- [ ] Run spotless check
- [ ] Run tests
- [ ] Update metalava's signature text files

### CODE REVIEW

File: datalayer/sample/wear/src/main/res/values/wear_keep.xml
Comment (yschimke) [inline datalayer/sample/wear/src/main/res/values/wear_keep.xml:1]: See https://b.corp.google.com/issues/348688201#comment9 I believe this needs to move from res/values to res/raw

File: docs/datalayer-helpers-guide.md [hunk 1]
```diff
@@ -55,6 +55,11 @@ phone.
     </string-array>
     </resources>
     ```
+   and a `wear_keep.xml`
```
Comment (yschimke) [inline docs/datalayer-helpers-guide.md:58]: Maybe a full path including res/raw?

File: docs/datalayer-helpers-guide.md [hunk 2]
```diff
@@ -30,7 +30,7 @@ phone.
 
    Add a `wear.xml` file in the `res/values` folder with the following content:
```
Comment (yschimke) [inline docs/datalayer-helpers-guide.md:31]: res/raw

**Comment by yschimke** [comment]
cc @kul3r4 FYI

File: datalayer/sample/wear/src/main/res/values/wear_keep.xml
Comment (oas004) [reply datalayer/sample/wear/src/main/res/values/wear_keep.xml:1]: Don't have access to that link :( Could u provide the information from that?

File: docs/datalayer-helpers-guide.md (hunk 1 above)
Comment (oas004) [reply docs/datalayer-helpers-guide.md:58]: Good idea! I'll change that :)

File: datalayer/sample/wear/src/main/res/values/wear_keep.xml
Comment (oas004) [reply datalayer/sample/wear/src/main/res/values/wear_keep.xml:1]: I can just move it :)

Comment (oas004) [reply datalayer/sample/wear/src/main/res/values/wear_keep.xml:1]: Moved it in [b0756e4](https://github.com/google/horologist/pull/2385/commits/b0756e4e8db3d566fade5e406dfc01335389b522)

Comment (yschimke) [reply datalayer/sample/wear/src/main/res/values/wear_keep.xml:1]: Same link you provided

File: docs/datalayer-helpers-guide.md [hunk 3]
```diff
@@ -28,9 +28,9 @@ phone.
 
 2. Add the capabilities
 
-   Add a `wear.xml` file in the `res/values` folder with the following content:
+   Add a `wear.xml` file in the `res/raw` folder with the following content:
```
Comment (yschimke) [inline docs/datalayer-helpers-guide.md:31]: Can you mention that it needs to be res/raw, you could link to the bug.

File: datalayer/sample/wear/src/main/res/values/wear_keep.xml
Comment (oas004) [reply datalayer/sample/wear/src/main/res/values/wear_keep.xml:1]: Ah okay, sorry it was just asking me to provide google sso so was a bit confused 😅

File: docs/datalayer-helpers-guide.md (hunk 3 above)
Comment (oas004) [reply docs/datalayer-helpers-guide.md:31]: Yes, good idea! I can add that :) Thanks!

File: docs/datalayer-helpers-guide.md [hunk 4]
```diff
@@ -28,9 +28,10 @@ phone.
 
 2. Add the capabilities
 
-   Add a `wear.xml` file in the `res/values` folder with the following content:
+   Add a `wear.xml` file in the `res/raw` folder with the following content. It needs to be
+   `res/raw` folder due to https://issuetracker.google.com/issues/348688201
```
[the remark this answers is not in the document]
Comment (yschimke) [reply docs/datalayer-helpers-guide.md:32]: Sounds good.

**Comment by kul3r4** [comment]
Shall I also change https://github.com/android/wear-os-samples/blob/main/DataLayer/Wearable/src/main/res/values/wear.xml?

**Comment by yschimke** [comment]
@oas004 Good catch by Chiara, can you update that also in this PR? Thanks for your patience with changes bouncing back and forth.

**Comment by oas004** [comment]
> @oas004 Good catch by Chiara, can you update that also in this PR? Thanks for your patience with changes bouncing back and forth.

No worries at all! Thank you for thorough reviews! 🙌  The change that she is referring to isn't that in wear-os sample repository? I can update that as well, but I'm not sure it is possible in this PR? Maybe I'm misunderstanding something?

**Comment by kul3r4** [comment]
> No worries at all! Thank you for thorough reviews! 🙌 The change that she is referring to isn't that in wear-os sample repository? I can update that as well, but I'm not sure it is possible in this PR? Maybe I'm misunderstanding something?

Yes it's a different repository, happy to review if you want to send a PR there.

**Comment by yschimke** [comment]
@oas004 @kul3r4 oooofffff, sorry.

**Pull request merged**